astronomer/providers/microsoft/azure/triggers/wasb.py

```diff
@@ -0,0 +1,148 @@
+    existence of the given blob in the provided container.
+
+    :param container_name: name of the container in which the blob should be searched for
+    :param blob_name: name of the blob to check existence for
+    :param wasb_conn_id: the connection identifier for connecting to Azure WASB
+    :param poll_interval:  polling period in seconds to check for the status
+    :param public_read: whether an anonymous public read access should be used. Default is False
+    """
+
+    def __init__(
+        self,
+        container_name: str,
+        blob_name: str,
+        wasb_conn_id: str = "wasb_default",
+        public_read: bool = False,
+        poll_interval: float = 5.0,
+    ):
+        super().__init__()
+        self.container_name = container_name
+        self.blob_name = blob_name
+        self.wasb_conn_id = wasb_conn_id
+        self.poll_interval = poll_interval
+        self.public_read = public_read
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """Serializes WasbBlobSensorTrigger arguments and classpath."""
+        return (
+            "astronomer.providers.microsoft.azure.triggers.wasb.WasbBlobSensorTrigger",
+            {
+                "container_name": self.container_name,
+                "blob_name": self.blob_name,
+                "wasb_conn_id": self.wasb_conn_id,
+                "poll_interval": self.poll_interval,
+                "public_read": self.public_read,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """Makes async connection to Azure WASB and polls for existence of the given blob name."""
+        blob_exists = False
+        hook = WasbHookAsync(wasb_conn_id=self.wasb_conn_id, public_read=self.public_read)
+        try:
+            async with hook.blob_service_client:
+                while not blob_exists:
+                    blob_exists = await hook.check_for_blob(
+                        container_name=self.container_name,
+                        blob_name=self.blob_name,
+                    )
+                    if blob_exists:
+                        message = f"Blob {self.blob_name} found in container {self.container_name}."
+                        yield TriggerEvent({"status": "success", "message": message})
+                    else:
+                        message = (
+                            f"Blob {self.blob_name} not available yet in container {self.container_name}."
+                            f" Sleeping for {self.poll_interval} seconds"
+                        )
+                        self.log.info(message)
+                        await asyncio.sleep(self.poll_interval)
+        except Exception as e:
+            yield TriggerEvent({"status": "error", "message": str(e)})
+
+
+class WasbPrefixSensorTrigger(BaseTrigger):
+    """
+    WasbPrefixSensorTrigger is fired as deferred class with params to run the task in trigger worker to check for
```

Comment:
Consider breaking this into two sentences for better readability

```suggestion
    WasbPrefixSensorTrigger is fired as a deferred class with params to run the task in trigger worker.
```